krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,238 @@
+func NewFromYAML(b []byte) (Interface, error) {
+	o, err := fn.ParseKubeObject(b)
+	return &itfce{
+		o: o,
+	}, err
+}
+
+// NewFromGoStruct creates a new parser interface
+// It expects a interface go struct
+func NewFromGoStruct(x *nephioreqv1alpha1.Interface) (Interface, error) {
+	b, err := yaml.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+	return NewFromYAML(b)
+}
+
+type itfce struct {
+	o *fn.KubeObject
+}
+
+// GetKubeObject returns the present kubeObject
+func (r *itfce) GetKubeObject() *fn.KubeObject {
+	return r.o
+}
+
+func (r *itfce) GetGoStruct() (*nephioreqv1alpha1.Interface, error) {
+	x := &nephioreqv1alpha1.Interface{}
+	if err := yaml.Unmarshal([]byte(r.o.String()), x); err != nil {
+		return nil, err
+	}
+	return x, nil
+}
+
+func (r *itfce) GetAttachmentType() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "attachmentType")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) GetCNIType() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "cniType")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) GetNetworkInstanceName() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "networkInstance", "name")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) SetAttachmentType(s string) error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.AttachmentTypeNone):
+	case string(nephioreqv1alpha1.AttachmentTypeVLAN):
+	default:
+		return fmt.Errorf("unknown attachmentType")
+	}
+
+	if err := r.o.SetNestedField(s, "spec", "attachmentType"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) DeleteAttachmentType() error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	_, err := r.o.RemoveNestedField("spec", "attachmentType")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) SetCNIType(s string) error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.CNITypeIPVLAN):
+	case string(nephioreqv1alpha1.CNITypeSRIOV):
+	case string(nephioreqv1alpha1.CNITypeMACVLAN):
+	default:
+		return fmt.Errorf("unknown cniType")
+	}
+
+	if err := r.o.SetNestedField(s, "spec", "cniType"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) DeleteCNIType() error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	_, err := r.o.RemoveNestedField("spec", "cniType")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) SetNetworkInstanceName(s string) error {
+	if r.o == nil {
```

Comment:
Similar to getters , this block of code is same for setters and can be refactored to a common method. Call the validate method and then call the common setter method.

Reply:
fixed